src/Common/src/TypeSystem/Common/TypeSystemHelpers.cs

```diff
@@ -79,6 +79,13 @@ static private MethodDesc FindMethodOnExactTypeWithMatchingTypicalMethod(this Ty
                     return methodToExamine;
             }
 
+            // TODO: can we enumerate just the virtual ones?
+            foreach (MethodDesc methodToExamine in type.GetAllVirtualMethods())
```

Comment:
So, the right fix is to actually to use TypeSystemContext.GetMethodForInstantiatedType instead of looping in the first place.

---
In reply to: [58146161](https://github.com/dotnet/corert/pull/1068#discussion_r58146161) [](ancestors = 58146161)